src/payment/strategies/masterpass/masterpass-payment-strategy.ts

```diff
@@ -0,0 +1,127 @@
+import { PaymentStrategy } from '../';
+import {
+    PaymentActionCreator,
+    PaymentInitializeOptions,
+    PaymentMethodActionCreator,
+    PaymentRequestOptions
+} from '../../';
+import { Checkout, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType
+} from '../../../common/error/errors';
+import { StoreConfig } from '../../../config';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentMethod from '../../payment-method';
+
+import { MasterpassCheckoutCallback, MasterpassCheckoutOptions } from './masterpass';
+import MasterpassScriptLoader from './masterpass-script-loader';
+
+export default class MasterpassPaymentStrategy extends PaymentStrategy {
+    private _paymentMethod?: PaymentMethod;
+    private _paymentGateway?: string;
+
+    constructor(
+        store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _masterpassScriptLoader: MasterpassScriptLoader
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId, masterpass: masterpassOptions } = options;
+
+        if (!masterpassOptions) {
+            throw new InvalidArgumentError('Unable to initialize payment because masterpass options is missing');
+        }
+
+        const state = this._store.getState();
+        this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+        return Promise.resolve(this._hasPaymentInfo()).then(hasPaymentInfo => {
+            if (!hasPaymentInfo) {
+                return this._masterpassClientSetup().then(checkoutCallback => checkoutCallback());
+            }
+
+            this._paymentGateway = masterpassOptions.gateway;
+
+            if (masterpassOptions.onPaymentSelect) {
+                masterpassOptions.onPaymentSelect();
```

Comment:
I'm not sure why you're calling `onPaymentSelect` here because I'm pretty sure at this point the shopper hasn't selected a payment option from their wallet.

Reply:
i removed onPaymentSelect, at some point of the development was necessary but with the latest changes on ng-checkout isn't required anymore